rpcs3/Input/pad_thread.cpp

```diff
@@ -375,7 +383,80 @@ void pad_thread::operator()()
 			}
 		}
 
-		thread_ctrl::wait_for(g_cfg.io.pad_sleep);
+		if (m_resume_emulation_flag)
+		{
+			m_resume_emulation_flag = false;
+
+			Emu.BlockingCallFromMainThread([]()
+			{
+				Emu.Resume();
+			});
+		}
+
+		u64 pad_sleep = g_cfg.io.pad_sleep;
+
+		if (Emu.IsPaused())
+		{
+			pad_sleep = std::max<u64>(pad_sleep, 50'000);
+
+			u64 timestamp = get_system_time();
+			u32 pressed_mask = 0;
+
+			for (usz i = 0; i < m_pads.size(); i++)
+			{
+				const auto& pad = m_pads[i];
+
+				if (!(pad->m_port_status & CELL_PAD_STATUS_CONNECTED))
+					continue;
+
+				for (auto& button : pad->m_buttons)
+				{
+					if (button.m_offset == CELL_PAD_BTN_OFFSET_DIGITAL1 && button.m_outKeyCode == CELL_PAD_CTRL_START && button.m_pressed)
+					{
+						pressed_mask |= 1u << i;
+						break;
+					}
+				}
+			}
+
+			m_mask_start_press_to_unpause &= pressed_mask;
+
+			if (!pressed_mask || timestamp - m_track_start_press_begin_timestamp >= 1'200'000)
+			{
+				m_track_start_press_begin_timestamp = timestamp;
+
+				if (std::exchange(m_mask_start_press_to_unpause, u32{umax}))
+				{
+					m_mask_start_press_to_unpause = 0;
+					m_track_start_press_begin_timestamp = 0;
+
+					rsx::overlays::queue_message("Unpausing...!"s, 2'500'000);
```

Comment:
Why is this whole bloat needed here?
Doesn't the player see immediately that the game is running?
Why add this complex native ui section in the pad thread?

Reply:
So the player would have time to prepare for the gameplay. You need to remember that savestating and pausing can occur at any given momement of gameplay and this makes it comfortable to do it at nearly every momemnt even in the middle of a boss fight.

Reply:
Then it should probably just set m_info.ignore_input to true and skip the = false as long as the unpause timeout is not reached.
That section was added to hide button presses after a native ui dialog was closed

Reply:
Although then you probably die in dark souls if you try to move the sticks before it is unpaused